commands/project_script.go

```diff
@@ -0,0 +1,88 @@
+package commands
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/phase2/rig/util"
+)
+
+// ProjectScript wrapps the evaluation of project scripts.
+// It mimics command struct except with unexported values.
+type ProjectScript struct {
+	out    *util.RigLogger
+	config *ProjectConfig
+}
+
+// Run takes a Script configuration and executes it per the definition of
+// the project script and bonus arguments from the extra parameter.
+// Commands are run from the directory context of the project if available.
+// This also supports follow-up user interaction.
+func (p *ProjectScript) Run(script *Script, extra []string) int {
+	p.out.Verbose("Initializing project script '%s': %s", script.ID, script.Description)
+	p.addCommandPath()
+	dir := p.GetWorkingDirectory()
+	shellCmd := p.GetCommand(script.Run, extra, dir)
+	shellCmd.Env = append(os.Environ(), "RIG_POWER_USER_MODE=1")
+	p.out.Verbose("Evaluating Script '%s'", script.ID)
```

Comment:
Seems like the previous 6 lines are the same as in `Capture` and should be DRY'd to something like

`util.PassthruCommand(p.GenerateCommand())` or something like it.